push-agents/apns-agent/src/main/java/org/aerogear/push/apns/helper/MessageHolderWithTokens.java

```diff
@@ -0,0 +1,60 @@
+package org.aerogear.push.apns.helper;
+
+import org.jboss.aerogear.unifiedpush.api.FlatPushMessageInformation;
+import org.jboss.aerogear.unifiedpush.api.Variant;
+import org.jboss.aerogear.unifiedpush.api.iOSVariant;
+import org.jboss.aerogear.unifiedpush.message.UnifiedPushMessage;
+
+import java.io.Serializable;
+import java.util.Collection;
+
+public class MessageHolderWithTokens  {
+
+    private static final long serialVersionUID = -7955411139315335655L;
+
+    private int serialId;
+    private iOSVariant variant;
+    private Collection<String> deviceTokens;
+    private FlatPushMessageInformation pushMessageInformation;
+    private InternalUnifiedPushMessage unifiedPushMessage;
+
+
+    public MessageHolderWithTokens() {
+
+    }
+
+    public MessageHolderWithTokens(FlatPushMessageInformation pushMessageInformation, UnifiedPushMessage unifiedPushMessage, Variant variant, Collection<String> deviceTokens, int serialId) {
+        this.pushMessageInformation = pushMessageInformation;
+        this.unifiedPushMessage = (InternalUnifiedPushMessage) unifiedPushMessage;
+        if (!(deviceTokens instanceof Serializable)) {
+            throw new IllegalArgumentException("deviceTokens must be a serializable collection");
+        }
+        this.variant = (iOSVariant) variant;
+        this.deviceTokens = deviceTokens;
```

Comment:
This is a potential issue, since exposes a mutable internal object. Would be better to copy it.